### PR TITLE
US001 - View Job Roles

### PR DESCRIPTION
**Pull request by hannahmorgan24**

### CODE REVIEW

File: src/controllers/JobRoleController.ts
Comment (RaggedyRach) [inline src/controllers/JobRoleController.ts:1]: Hi Hannah, I think we may want this in a try catch. 
image

**Review by RaggedyRach — COMMENTED** [review]
One possible change a try catch in the controller. 
Not sure if linting on front end will want spaces at end of each file like the back end. I will try to get it on the UI build so we can check it

File: views/jobRoleList.html
Comment (RaggedyRach) [inline views/jobRoleList.html:1]: oh and maybe change this to Open Job Roles List

File: src/controllers/JobRoleController.ts
Comment (shaunganley) [reply src/controllers/JobRoleController.ts:1]: agreed

File: src/models/Band.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export enum Band {
```
Comment (shaunganley) [inline src/models/Band.ts]: don't want to hardcode bands, keep this as a string

File: src/models/Capability.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export enum Capability {
```
Comment (shaunganley) [inline src/models/Capability.ts]: don't want to hardcode capability, keep this as a string

File: test/unit/services/JobRoleServiceTests.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+/*
+Unit test for getJobRoles response
+*/
+
+const mock = new MockAdapter(axios);
+
+describe('JobRoleService', function () {
+    describe('getAllJobRoles', function () {
+      it('should return job roles from response', async () => {
```
Comment (shaunganley) [inline test/unit/services/JobRoleServiceTests.ts]: add a unit test for 500 error from axios

File: views/jobRoleList.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <h1>Job Roles</h1>
+    <table>
+        <tr>
+            <th>Job Role Name</th>
+            <th>Job Role Location</th>
+            <th>Job Role Capability</th>
+            <th>Job Role Band</th>
+            <th>Job Role Closing Date</th>
+        </tr>
+        {% for openJobRole in openJobRoles | sort(attribute='jobRoleID') %}
```
Comment (shaunganley) [inline views/jobRoleList.html]: jobRoleID doesn't exist in your model

**Review by shaunganley — CHANGES_REQUESTED** [review]

**Review by shaunganley — APPROVED** [review]

**Review by RaggedyRach — APPROVED** [review]

**Review by Brigid481 — APPROVED** [review]
looks good Rachael

**Review by RaggedyRach — APPROVED** [review]

**Pull request merged**